lte/gateway/c/oai/tasks/sgw/pgw_handlers.c

```diff
@@ -418,3 +419,129 @@ static void get_session_req_data(
   data->pvi = qos->pvi;
   data->qci = qos->qci;
 }
+
+//-----------------------------------------------------------------------------
+
+uint32_t pgw_handle_nw_initiated_bearer_actv_req(
+  Imsi_t *imsi,
+  ebi_t lbi,
+  traffic_flow_template_t *ul_tft,
+  traffic_flow_template_t *dl_tft,
+  bearer_qos_t *eps_bearer_qos)
+{
+  OAILOG_FUNC_IN(LOG_PGW_APP);
+  MessageDef *message_p = NULL;
+  uint32_t i = 0;
+  uint32_t rc = RETURNok;
+  hash_table_ts_t *hashtblP = NULL;
+  uint32_t num_elements = 0;
+  s_plus_p_gw_eps_bearer_context_information_t *spgw_ctxt_p = NULL;
+  hash_node_t *node = NULL;
+  itti_s5_nw_init_actv_bearer_request_t
+    *itti_s5_actv_bearer_req = NULL;
+
+  OAILOG_INFO(
+    LOG_PGW_APP,
+    "Received Create Bearer Req from PCRF with IMSI %s\n",
+    imsi->digit);
+
+  teid_t pgw_u_teid = sgw_get_new_s1u_teid();
+  message_p =
+    itti_alloc_new_message(TASK_SPGW_APP,
+      S5_NW_INITIATED_ACTIVATE_BEARER_REQ);
+  if (message_p == NULL) {
+    OAILOG_ERROR(
+    LOG_PGW_APP,
+    "itti_alloc_new_message failed for"
+    "S5_NW_INITIATED_ACTIVATE_DEDICATED_BEARER_REQ\n");
+    OAILOG_FUNC_RETURN(LOG_PGW_APP, RETURNerror);
+  }
+  itti_s5_actv_bearer_req =
+    &message_p->ittiMsg.s5_nw_init_actv_bearer_request;
+  //Send ITTI message to SGW
+  memset(
+    itti_s5_actv_bearer_req,
+    0,
+    sizeof(itti_s5_nw_init_actv_bearer_request_t));
+
+  //Copy Bearer QoS
+  memcpy(
+    &itti_s5_actv_bearer_req->eps_bearer_qos,
+    eps_bearer_qos,
+    sizeof(bearer_qos_t));
+  //Copy TFT
+  memcpy(
+    &itti_s5_actv_bearer_req->tft,
+    ul_tft,
+    sizeof(traffic_flow_template_t));
+  //Assign TEID
+  itti_s5_actv_bearer_req->S5_U_pgw_teid = pgw_u_teid;
```

Comment:
What is the use of itti_s5_actv_bearer_req->S5_U_pgw_teid ?

Reply:
Removed. I had added a place holder for PGW TEID in the ITTI message for future use as it be  will sent on S5 interface in Create Bearer Request as per 29.274